### PR TITLE
adding line_number validation code

### LINKED ISSUE #3741: re-test and re-evaluate line number-filtering loading performance and add data format validation in line number
What we are after:
As pat help identified a line_number format/value issue when I test sched_a and sched_b endpoints, I need to re-test and re-evaluate line_number related loading performance. We also need to add the line number data format validator and make it more robust.

Acceptance Criteria:
- adding a line_number format validator and fail the endpoints when a in-valid values specified.
- re-test and re-evaluate line_number related loading performance
- add unit tests if necessary.

### PR DESCRIPTION
**Pull request by qqss88**
## Summary (required)
adding line_number validation code and add line_number back to sched_a secondary index list
- Resolves #3741

_Include a summary of proposed changes._
1. line_number validation code added to both sched_a and sched_b
2. unit tests updated
## How to test the changes locally

1. start local server and connect to stage db
./manage.py runserver

2. the local requests should give you error messages:

request:
http://127.0.0.1:5000/v1/schedules/schedule_b/?per_page=50&sort_null_only=false&sort_hide_null=false&api_key=DEMO_KEY&two_year_transaction_period=2018&two_year_transaction_period=2016&two_year_transaction_period=2014&line_number=F3X23
response:
{
    "message": "Invalid line_number detected.",
    "status": 400
}

request:
http://127.0.0.1:5000/v1/schedules/schedule_a/?per_page=50&sort_null_only=false&sort_hide_null=false&api_key=DEMO_KEY&two_year_transaction_period=2018&two_year_transaction_period=2016&two_year_transaction_period=2014&line_number=F3X11AI
response:
{
    "message": "Invalid line_number detected.",
    "status": 400
}

3. all unit tests should pass

-

## Impacted areas of the application
List general components of the application that this PR will affect:

-  

## Related PRs
List related PRs against other branches:

branch | PR
------ | ------
fix/other_pr | [link]()
feature/other_pr | [link]()

### CODE REVIEW

**Comment by fec-jli** [comment]
Thanks good work.  line-number validation is good. 
I test a couple of line-numbers (first time call, without cache). 
1)F3-16 took 3min,  
2)F3-15 took 3 min.  
3)F3X-18 never return result.  
my local points to dev rep1, inside fec building. 
maybe on prod-rep1 better.

File: webservices/resources/sched_a.py [hunk 1]
```diff
@@ -118,10 +119,16 @@ def build_query(self, **kwargs):
...
             query = query.filter_by(sub_id=int(kwargs.get('sub_id')))
         if kwargs.get('line_number'):
+            # line_number is a composite value of 'filing_form-line_number'
             if len(kwargs.get('line_number').split('-')) == 2:
                 form, line_no = kwargs.get('line_number').split('-')
                 query = query.filter_by(filing_form=form.upper())
                 query = query.filter_by(line_number=line_no)
+            else:
+                raise exceptions.ApiError(
+                    "Invalid line_number detected.",
```
Comment (lbeaufort) [inline webservices/resources/sched_a.py]: Can we tell the user what the required line number format is?

Comment (qqss88) [reply webservices/resources/sched_a.py]: updated. thx.

File: webservices/exceptions.py [hunk 2]
```diff
@@ -1,3 +1,9 @@
+
+LINE_NUMBER_ERROR = """
+Invalid line_number detected. A valid line_number is using the following format:
+'FORM-LINENUMBER'.  Check the api doc for more info.
```
Comment (lbeaufort) [inline webservices/exceptions.py]: I don't think `Check the api doc for more info.` adds value here, can you remove it?

Comment (lbeaufort) [reply webservices/exceptions.py]: Maybe instead we could give them an example, like the docs do? From the docs: `For example an argument such as `F3X-16` would filter down to all entries from form `F3X` line number `16`.`

Comment (qqss88) [reply webservices/exceptions.py]: Sure - that definitely adds value. There is a minor issue for the docs too: F3X-16 is only an example of Schedule_a, but it is used for both schedule_a and schedule_b. So it is a little confusing on this part.

File: webservices/resources/sched_b.py [hunk 3]
```diff
@@ -8,10 +8,13 @@
 from webservices.common import models
 from webservices.common import views
 from webservices.common.views import ItemizedResource
-
+from webservices import exceptions
 """
 two years restriction removed from schedule_b. For details, refer:
 https://github.com/fecgov/openFEC/issues/3595
+"""
+LINE_NUMBER_ERROR = """
```
Comment (lbeaufort) [inline webservices/resources/sched_b.py]: Can we remove this `LINE_NUMBER_ERROR = """`?

**Pull request merged**